### DISCUSSION TITLE
Wonderful Plugin!! and a few bugs...

### DISCUSSION THREAD

**Discussion by panhardleft** [rerdavies/pipedal#249] [General]
I have just discovered Pipedal and I am thrilled: super stable and powerful, great GUI, congratulations! And thank you very much!!
Unfortunately I also discovered a few bugs:
1. After installation, no Nam profiles or IRs can be uploaded, probably because permissions are missing. But if you delete the folder /var/pipedal/audio_uploads/, you can upload some via the web interface and the folder is recreated.
2. pipedal_latency_test needs hw:<interface name> as a parameter, e.g. hw:AMS44 and not AMS44

But other than that it's fantastic!!

**Comment by rerdavies** [comment]
Thanks for your problem reports.

I will check into 1  right away. 

And is issue 2 a docs issue or an issue with the help text?  Or both?

**Reply by panhardleft** [reply]
That depends on you - I think it's a very small error in parsing the string returned by aplay -l (or something like that - I'm more of a noob unfortunately). You just have to put "hw:" in front of the interface name and then pipedal_latency_test will work as expected

**Comment by rerdavies** [comment]
:) I knew you had to add hw: to the name. But if docs or help mislead you
into thinking you didn't, I'd like to fix that.

**Reply by panhardleft** [reply]
ah! Got it! Yes the I was mislead by the help text, mainly by your example: "pipedal_latency_test M2"

**Comment by rerdavies** [comment]
Sorry for the slow reponse on this. I lost one of my development machines, so I've been rebuild systems, and doing some long-overdue housekeeping. (No sources lost, thankfully).

The permissions issue is a bug. It will be fixed in an imminent release. In the meantime, the easiest workaround is probably the following:

Run the following command:

    pipedalconfig --install

This will non-destructively run re-run the installation procedures, and should correct the permissions on the audio_uploads folder, allowing you to upload samples and model files. (Deleting the audio_uploads directory also works).
